### PR TITLE
Upgrade Jest from 29 to 30

### PR DESCRIPTION
**Pull request by ravikiranvm**
Fixes OPS-4816.

Supersedes #2485. Dependabot's jest group bump left `jest-environment-node` and `jest-environment-jsdom` on 29.7.0, so the environments handed the Jest 30 runtime a Jest 29 module mocker and every server-api suite crashed on startup with `TypeError: this._moduleMocker.clearMocksOnScope is not a function`. Jest 30 is a major release that also needs test code changes, so the upgrade is finished here by hand.

## Additional Notes

Builds on the Dependabot commits (jest 30.4.2, babel-jest 30.4.1, @types/jest 30.0.0, ts-jest 29.4.12) and adds:

- **Bump `jest-environment-node` and `jest-environment-jsdom` to 30.4.1** so the tree resolves a single `jest-mock`. This also moves the UI tests from jsdom 20 to jsdom 26.
- **Rename `testPathPattern` to `testPathPatterns`** in the three server-api test targets. Jest 30 renamed the option and silently ignored the old name, so the unit target was running integration suites too.
- **Replace matcher aliases removed in Jest 30** (`toThrowError`, `toBeCalled`, `toBeCalledWith`, `toBeCalledTimes`) with `toThrow` and the `toHaveBeenCalled*` forms across 23 test files.
- **Make the OAuth consent hook test jsdom 26 compatible.** jsdom 26 marks `window.location` non-configurable, so the test can no longer replace it. The redirect now goes through a one-line `navigateToExternalUrl` helper that the test mocks at the module boundary. Hook behavior is unchanged.

No product code changes beyond that helper extraction.

## Testing Checklist

Check all that apply:

- [x] I tested the feature thoroughly, including edge cases

- [x] I verified all affected areas still work as expected

- [x] Automated tests were added/updated if necessary

- [x] Changes are backwards compatible with any existing data, otherwise a migration script is provided

Ran locally on Jest 30:

| Target | Result |
| --- | --- |
| `nx test-unit server-api` | 97 suites, 962 tests passed, only `test/unit` files ran |
| `nx test-integration-ce server-api` | 18 suites, 148 tests passed |
| `nx test-integration-cloud server-api` | 10 suites, 67 tests passed |
| `nx test react-ui` | 49 suites, 546 tests passed |
| `nx test ui-components` | 13 suites, 203 tests passed |
| `nx run-many --target=test` (blocks, engine, openops, shared) | passed |
| lint on react-ui, server-api, openops, blocks-anodot | clean |

Two block suites fail on my machine for reasons unrelated to this change and pass in CI: `blocks-date-helper` assumes a three-letter short month and my locale is en-IN ("Sept"), and `blocks-azure` expects the `AZURE_EXTENSION_DIR` env var that GitHub runners set.

## Visual Changes (if applicable)

None.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -367,7 +367,7 @@
     "@vitest/eslint-plugin": "1.6.27",
     "@vitest/ui": "1.6.1",
     "autoprefixer": "10.5.0",
-    "babel-jest": "29.7.0",
+    "babel-jest": "30.4.1",
```
[the remark this answers is not in the document]
Comment (ravikiranvm) [reply package.json:370]: There is no v30.4.2 for babel-jest. Let's wait for dependabot to pickup latest version 30.5.1

**Review by MarceloRGonc — APPROVED** [review]
